### PR TITLE
Fix failing unit-test test_wsireader

### PR DESCRIPTION
**Pull request by hjmjohnson**
Conversion of units when the unit if '' caused the test to fail.

```bash
pytest -k tiff
```

FAILED tests/test_wsireader.py::TestTiffFile::test_resolution_mpp_0__home_johnsonhj_src_MONAI_tests_testing_data_temp_wsi_generic_tiff_tiff - ValueError: Currently, it only supports length conversion but `` is given.

FAILED tests/test_wsireader.py::TestTiffFile::test_resolution_mpp_0__home_johnsonhj_src_MONAI_tests_testing_data_temp_wsi_generic_tiff_tiff - AttributeError: 'ConvertUnits' object has no attribute 'conversion_factor'

### Description

Fixes a failing test.  

### Types of changes
- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.

### CODE REVIEW

**Comment by KumoLiu** [comment]
Hi @hjmjohnson, thanks for the PR, could you please show more details when will the test failed such a small piece of code and also include the test case in the unittests? Thanks!

**Comment by hjmjohnson** [comment]
> Hi @hjmjohnson, thanks for the PR, could you please show more details when will the test failed such a small piece of code and also include the test case in the unittests? Thanks!

This is the failing unit test output from the original code:
```
FAILED tests/test_wsireader.py::TestTiffFile::test_resolution_mpp_0__home_johnsonhj_src_MONAI_tests_testing_data_temp_wsi_generic_tiff_tiff - ValueError: Currently, it only supports length conversion but `` is given.

FAILED tests/test_wsireader.py::TestTiffFile::test_resolution_mpp_0__home_johnsonhj_src_MONAI_tests_testing_data_temp_wsi_generic_tiff_tiff - AttributeError: 'ConvertUnits' object has no attribute 'conversion_factor'
```

I am trying to get full passing of all the monai tests on my computer, and these are the failures I came across.

File: monai/utils/misc.py [hunk 1]
```diff
@@ -841,7 +841,7 @@ def _calculate_conversion_factor(self):
             return 1.0
         input_power = self._get_unit_power(self.input_unit)
         target_power = self._get_unit_power(self.target_unit)
-        self.conversion_factor = 10 ** (input_power - target_power)
+        return 10 ** (input_power - target_power)
```
Comment (hjmjohnson) [inline monai/utils/misc.py:844]: NOTE: In previous code the "self.conversion_factor" variable was never created or set if "if self.input_unit == self.target_unit:"

**Comment by hjmjohnson** [comment]
```bash
git checkout dev
git reset origin/dev --hard
pytest -k tiff
```

```txt
============================================================================================= FAILURES =============================================================================================
____________________________________________ TestTiffFile.test_resolution_mpp_0__home_johnsonhj_src_MONAI_tests_testing_data_temp_wsi_generic_tiff_tiff ____________________________________________

a = (<tests.test_wsireader.TestTiffFile testMethod=test_resolution_mpp_0__home_johnsonhj_src_MONAI_tests_testing_data_temp_wsi_generic_tiff_tiff>,), kw = {}

    @wraps(func)
    def standalone_func(*a, **kw):
>       return func(*(a + p.args), **p.kwargs, **kw)

.venv/lib/python3.12/site-packages/parameterized/parameterized.py:620: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/test_wsireader.py:593: in test_resolution_mpp
    mpp = reader.get_mpp(img_obj, level)
monai/data/wsi_reader.py:604: in get_mpp
    return self.reader.get_mpp(wsi, level)
monai/data/wsi_reader.py:1105: in get_mpp
    convert_to_micron = ConvertUnits(unit, "micrometer")
monai/utils/misc.py:808: in __init__
    self.input_unit, input_base = self._get_valid_unit_and_base(input_unit)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <monai.utils.misc.ConvertUnits object at 0x79d5f3340380>, unit = ''

    def _get_valid_unit_and_base(self, unit):
        unit = str(unit).lower()
        if unit in self.imperial_unit_of_length:
            return unit, "meter"
        for base_unit in self.base_units:
            if unit.endswith(base_unit):
                return unit, base_unit
>       raise ValueError(f"Currently, it only supports length conversion but `{unit}` is given.")
E       ValueError: Currently, it only supports length conversion but `` is given.

monai/utils/misc.py:826: ValueError
```

**Review by KumoLiu — APPROVED** [review]

**Pull request merged**